errai-common/src/main/java/org/jboss/errai/common/client/dom/elemental2/Elemental2DomUtil.java

```diff
@@ -0,0 +1,99 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.errai.common.client.dom.elemental2;
+
+import com.google.gwt.user.client.ui.RootPanel;
+import com.google.gwt.user.client.ui.Widget;
+import elemental2.dom.HTMLElement;
+import elemental2.dom.Node;
+import jsinterop.base.Js;
+
+/**
+ * Provides utility methods for interacting with the DOM.
+ * @author Guilherme Carreiro <ggomes@redhat.com>
+ */
+public class Elemental2DomUtil {
+
+    /**
+     * Detaches all element children from a node.
+     * @param node Must not be null.
+     * @return True iff any element children were detached by this call.
+     */
+    public static boolean removeAllElementChildren(final Node node) {
+
+        final boolean hadChildren = node.lastChild != null;
+
+        while (node.lastChild != null) {
+            node.removeChild(node.lastChild);
+        }
+
+        return hadChildren;
+    }
+
+    /**
+     * Appends the underlying {@link HTMLElement} of a {@link Widget} to another {@link HTMLElement},
+     * in a way that does not break GWT Widget events.
+     * @param parent The parent element that is appended to. Must not be null.
+     * @param child The child Widget, whose underlying HTML element will be
+     * appended to the parent. Must not be null.
+     */
+    public static void appendWidgetToElement(final HTMLElement parent, final Widget child) {
+
+        if (child.isAttached()) {
+            child.removeFromParent();
+        }
+
+        RootPanel.detachOnWindowClose(child);
+
+        HTMLElement newChild = asHTMLElement(child.getElement());
+        parent.appendChild(newChild);
+
+        attach(child);
+    }
+
+    /**
+     * Converts from {@link com.google.gwt.dom.client.Element} to {@link HTMLElement}.
+     * @param gwtElement The deprecated GWT Element
+     * @return The GWT Elemental2 HTMLElement
+     */
+    public static HTMLElement asHTMLElement(final com.google.gwt.dom.client.Element gwtElement) {
+        return Js.cast(gwtElement);
+    }
+
+    /**
+     * Converts from {@link org.jboss.errai.common.client.dom.HTMLElement} to {@link HTMLElement}.
+     * @param htmlElement The deprecated Errai HTMLElement
+     * @return The GWT Elemental2 HTMLElement
+     */
+    public static HTMLElement asHTMLElement(final org.jboss.errai.common.client.dom.HTMLElement htmlElement) {
+        return Js.cast(htmlElement);
+    }
+
+    private static void attach(final Widget child) {
+        final Elemental2Widget widget = Js.uncheckedCast(child);
+
+        widget.onAttach();
+    }
+
+    static class Elemental2Widget extends Widget {
+
+        @Override
+        protected void onAttach() {
+            super.onAttach();
```

Comment:
I'm not understanding the significance of overriding this method!? 

If it has some significance perhaps a comment in the code would be useful?

Reply:
This is a workaround to access the `onAttach` protected method without JSNI.

[Here](https://github.com/errai/errai/blob/master/errai-common/src/main/java/org/jboss/errai/common/client/dom/DOMUtil.java#L439-L441) the previous version :-)